.github/workflows/snapshot.yaml

```diff
@@ -0,0 +1,36 @@
+---
+
+name: Publish Snapshot
+
+on:
+  push:
+    branches:
+      - main
+      - '[0-9].x'
+  workflow_dispatch:
+
+jobs:
+
+  snapshot:
+    name: Deploy Snapshot
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up publishing to maven central
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+          cache: 'maven'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: mvn offline
+        run: |
+          mvn -q dependency:go-offline
+      - name: deploy
+        run: |
+          mvn --no-transfer-progress deploy
```

Comment:
Adding `-U -B` ensures latest dependency versions and uses batch-mode to skip interactive elements.

Reply:
On minor version branches, it might not be the behavior we want.